lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -350,13 +351,20 @@ export class UserAgentApplication {
     loginPopup(userRequest?: AuthenticationParameters): Promise<AuthResponse> {
         // validate request
         const request: AuthenticationParameters = RequestUtils.validateRequest(userRequest, true, this.clientId, Constants.interactionTypePopup);
+        const apiEvent: ApiEvent = this.telemetryManager.createAndStartApiEvent(request.correlationId, API_EVENT_IDENTIFIER.LoginPopup, this.logger);
 
         return new Promise<AuthResponse>((resolve, reject) => {
             this.acquireTokenInteractive(Constants.interactionTypePopup, true, request, resolve, reject);
-        }).catch((error: AuthError) => {
-            this.cacheStorage.resetTempCacheItems(request.state);
-            throw error;
-        });
+        })
+            .then((resp) => {
+                this.telemetryManager.stopAndFlushApiEvent(request.correlationId, apiEvent, true);
+                return resp;
+            })
+            .catch((error: AuthError) => {
+                this.cacheStorage.resetTempCacheItems(request.state);
+                this.telemetryManager.stopAndFlushApiEvent(request.correlationId, apiEvent, false, error.errorCode);
+                throw error;
+            });
```

Comment:
nit: In future, we should have a OnSuccess and OnFailure handlers for all Public APIs may be.

Reply:
Can you go into more detail what you mean here?

Reply:
For all public APIs, we now clear cache, update telemetry on both success and failure. I am suggesting we can do something like this:

```javascript
loginPopup().then( () => {
    onSuccess();
}).catch( () => {
    onFailure();
});

function onSuccess() {
    clearCache();
    updateTelemetry();
};
...
...
```